packages/app-desktop/gui/ProfileManagementScreen.tsx

```diff
@@ -0,0 +1,277 @@
+	);
+};
+
+class ProfileManagementScreenComponent extends React.Component<Props, State> {
+	public constructor(props: Props) {
+		super(props);
+		this.state = {
+			profiles: props.profileConfig.profiles,
+			filter: '',
+		};
+	}
+
+	public componentDidUpdate(prevProps: Props) {
+		if (prevProps.profileConfig !== this.props.profileConfig) {
+			this.setState({ profiles: this.props.profileConfig.profiles });
+		}
+	}
+
+	public onProfileRename = async (profile: Profile) => {
+		const newName = await dialogs.prompt(_('Profile name:'), '', profile.name);
+		if (newName === null || newName === undefined || newName === profile.name) {
+			return;
+		}
+
+		if (!newName.trim()) {
+			bridge().showErrorMessageBox(_('Profile name cannot be empty'));
+			return;
+		}
+
+		try {
+			const newProfiles = this.props.profileConfig.profiles.map(p => {
+				if (p.id === profile.id) {
+					return {
+						...p,
+						name: newName.trim(),
+					};
+				}
+				return p;
+			});
+
+			const newProfileConfig = {
+				...this.props.profileConfig,
+				profiles: newProfiles,
+			};
+
+			await saveProfileConfig(`${Setting.value('rootProfileDir')}/profiles.json`, newProfileConfig);
+			this.props.dispatch({
+				type: 'PROFILE_CONFIG_SET',
+				value: newProfileConfig,
+			});
+		} catch (error) {
+			logger.error(error);
+			bridge().showErrorMessageBox(error.message);
+		}
+	};
+
+	public onProfileDelete = async (profile: Profile) => {
+		const isCurrentProfile = profile.id === this.props.profileConfig.currentProfileId;
+		if (isCurrentProfile) {
+			bridge().showErrorMessageBox(_('The active profile cannot be deleted. Switch to a different profile and try again.'));
+			return;
+		}
+
+		const ok = bridge().showConfirmMessageBox(_('Delete profile "%s"?\n\nAll data, including notes, notebooks and tags will be permanently deleted.', profile.name), {
+			buttons: [_('Delete'), _('Cancel')],
+			defaultId: 1,
+		});
+		if (!ok) {
+			return;
+		}
+
+		const rootDir = Setting.value('rootProfileDir');
+		const profileDir = `${rootDir}/profile-${profile.id}`;
+
+		try {
+			await shim.fsDriver().remove(profileDir);
+			logger.info('Deleted profile directory: ', profileDir);
+		} catch (error) {
+			logger.error('Error deleting profile directory: ', error);
+			bridge().showErrorMessageBox(error.message);
+		}
+
+		try {
+			const newConfig = deleteProfileById(this.props.profileConfig, profile.id);
+			await saveProfileConfig(`${Setting.value('rootProfileDir')}/profiles.json`, newConfig);
+			this.props.dispatch({
+				type: 'PROFILE_CONFIG_SET',
+				value: newConfig,
+			});
+		} catch (error) {
+			logger.error(error);
+			bridge().showErrorMessageBox(error.message);
+		}
+	};
+
+	public onFilterUpdate = (updateEvent: React.ChangeEvent<HTMLInputElement>) => {
+		this.setState({ filter: updateEvent.target.value });
+	};
+
+	public render() {
+		const style = this.props.style;
+		const theme = themeStyle(this.props.themeId);
+
+		const rootStyle: CSSProperties = {
```

Comment:
Style: [Use SCSS for styling components in new code.](https://joplinapp.org/help/dev/spec/desktop_styling)